include/lbann/callbacks/callback_dump_weights.hpp

```diff
@@ -47,8 +47,8 @@ class lbann_callback_dump_weights : public lbann_callback {
   /**
    * @param basename The basename for writing files.
    */
-  lbann_callback_dump_weights(std::string basename, int batch_interval = 1) :
-    lbann_callback(batch_interval), m_basename(std::move(basename)) {}
+  lbann_callback_dump_weights(std::string basename) :
```

Comment:
Is there a reason why we need to dump weights every mini-batch?

Reply:
There's no change that results from this, it's purely cosmetic to clean up the interface. `lbann_callback` also defaults `batch_interval` to `1`, and in any case `dump_weights` only implements `on_train_begin` and `on_epoch_end`. See #289, which should be referenced somewhere in this PR.

Reply:
@ndryden , I mentioned in the PR that "there are few other minor changes that do not necessarily have to do with GAN", this is one of them. Yes, you are right I should have referenced #289 to avoid confusion.